src/pgp.rs

```diff
@@ -168,31 +182,63 @@ pub async fn pk_encrypt(
     private_key_for_signing: SignedSecretKey,
     compress: bool,
     anonymous_recipients: bool,
+    seipd_version: SeipdVersion,
 ) -> Result<String> {
     Handle::current()
         .spawn_blocking(move || {
             let mut rng = thread_rng();
 
-            let pkeys = public_keys_for_encryption
-                .iter()
-                .filter_map(select_pk_for_encryption);
-
             let msg = MessageBuilder::from_bytes("", plain);
-            let mut msg = msg.seipd_v1(&mut rng, SYMMETRIC_KEY_ALGORITHM);
-            for pkey in pkeys {
-                if anonymous_recipients {
-                    msg.encrypt_to_key_anonymous(&mut rng, &pkey)?;
-                } else {
-                    msg.encrypt_to_key(&mut rng, &pkey)?;
-                }
-            }
 
-            msg.sign(&*private_key_for_signing, Password::empty(), HASH_ALGORITHM);
-            if compress {
-                msg.compression(CompressionAlgorithm::ZLIB);
-            }
-
-            let encoded_msg = msg.to_armored_string(&mut rng, Default::default())?;
+            let encoded_msg = match seipd_version {
+                SeipdVersion::V1 => {
+                    let mut msg = msg.seipd_v1(&mut rng, SYMMETRIC_KEY_ALGORITHM);
+                    let pkeys = public_keys_for_encryption
+                        .iter()
+                        .filter_map(select_pk_for_encryption);
+
+                    for pkey in pkeys {
+                        if anonymous_recipients {
+                            msg.encrypt_to_key_anonymous(&mut rng, &pkey)?;
+                        } else {
+                            msg.encrypt_to_key(&mut rng, &pkey)?;
+                        }
+                    }
+
+                    msg.sign(&*private_key_for_signing, Password::empty(), HASH_ALGORITHM);
+                    if compress {
+                        msg.compression(CompressionAlgorithm::ZLIB);
+                    }
+
+                    msg.to_armored_string(&mut rng, Default::default())?
+                }
+                SeipdVersion::V2 => {
+                    let mut msg = msg.seipd_v2(
+                        &mut rng,
+                        SYMMETRIC_KEY_ALGORITHM,
+                        AeadAlgorithm::Ocb,
+                        ChunkSize::C8KiB,
+                    );
+                    let pkeys = public_keys_for_encryption
```

Comment:
`pkeys` can be defined above the `match` block to dedup the code a bit. IIUC `msg` has different types for SEIPD versions, so not sure the remaining code can be deduplicated